packages/react-provider/src/stories/FluentProviderFrame.stories.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { Button } from '@fluentui/react-button'; // codesandbox-dependency: @fluentui/react-button ^9.0.0-beta
+import { makeStyles } from '@fluentui/react-make-styles';
+import { FluentProvider } from '@fluentui/react-provider'; // codesandbox-dependency: @fluentui/react-provider ^9.0.0-beta
+import { teamsDarkTheme, webLightTheme } from '@fluentui/react-theme'; // codesandbox-dependency: @fluentui/react-theme ^9.0.0-beta
+
+const useStyles = makeStyles({
+  example: {
+    margin: '5px',
+  },
+  darkThemeBackground: theme => ({
+    backgroundColor: theme.colorBrandBackground,
+  }),
+});
+
+const PortalFrame: React.FunctionComponent<{
+  children: (externalDocument: Document) => React.ReactElement;
+}> = ({ children }) => {
+  const [frameRef, setFrameRef] = React.useState<HTMLIFrameElement | null>(null);
+
+  return (
+    <>
+      <iframe
+        ref={setFrameRef}
+        style={{ height: 300, width: 600, border: 0, padding: 20 }}
+        title="An example of Provider in iframe"
+      />
+      {frameRef &&
+        ReactDOM.createPortal(
+          children(frameRef.contentDocument as Document),
+          (frameRef.contentDocument as Document).body,
+        )}
+    </>
+  );
+};
+
+export const Frame = () => {
+  const styles = useStyles();
+
+  return (
+    <>
+      <FluentProvider theme={teamsDarkTheme}>
+        <div className={styles.example}>
+          <div className={styles.darkThemeBackground}>Teams Dark Theme before iframe</div>
+        </div>
+        <Button className={styles.example}>Teams Dark Theme before iframe</Button>
+        <PortalFrame>
+          {externalDocument => {
+            return (
+              <>
+                <FluentProvider targetDocument={externalDocument}>
+                  <div>
+                    <div className={styles.example}>No theme within iframe</div>
+                    <Button className={styles.example}>No theme within iframe</Button>
+                  </div>
+                </FluentProvider>
+                <FluentProvider targetDocument={externalDocument} theme={webLightTheme}>
+                  <div>
+                    <div className={styles.example}>Web Light Theme within iframe</div>
+                    <Button className={styles.example}>Web Light Theme within iframe</Button>
+                  </div>
+                </FluentProvider>
```

Comment:
This example was outdated and broken as it referred  to the previous implementation 😥

It should actually show that it's possible to render into `iframe`s and this requires `RendererProvider`. I modified an example to make it properly working:

![image](https://user-images.githubusercontent.com/14183168/139839720-4dc51291-c6a3-4364-b39a-9ba6a5d62238.png)


```tsx
const useExampleStyles = makeStyles({
  button: {
    marginTop: '5px',
  },
  description: theme => ({
    backgroundColor: theme.colorBrandBackground2,
    color: theme.colorBrandForeground2,
    fontSize: '20px',
    border: '1px',
    borderRadius: '5px',
    padding: '5px',
  }),
});

const useProviderStyles = makeStyles({
  provider: {
    border: '1px',
    borderRadius: '5px',
    padding: '5px',
  },
});

type FrameRendererProps = {
  children: (externalDocument: Document, renderer: ReturnType<typeof createDOMRenderer>) => React.ReactElement;
};

const FrameRenderer: React.FunctionComponent<FrameRendererProps> = ({ children }) => {
  const [frameRef, setFrameRef] = React.useState<HTMLIFrameElement | null>(null);

  const contentDocument = frameRef ? (frameRef.contentDocument as Document) : undefined;
  const renderer = React.useMemo(() => createDOMRenderer(contentDocument), [contentDocument]);

  return (
    <>
      <iframe
        ref={setFrameRef}
        style={{ height: 100, width: 500, border: '3px dashed salmon', padding: 10 }}
        title="An example of Provider in iframe"
      />
      {contentDocument && ReactDOM.createPortal(children(contentDocument, renderer), contentDocument.body)}
    </>
  );
};

const Example: React.FC = props => {
  const styles = useExampleStyles();

  return (
    <>
      <div className={styles.description}>{props.children}</div>
      <Button className={styles.button}>A button</Button>
    </>
  );
};

export const Frame = () => {
  const styles = useProviderStyles();

  return (
    <>
      <FluentProvider className={styles.provider}>
        <Example>Content rendered outside iframe</Example>
      </FluentProvider>

      <FrameRenderer>
        {(externalDocument, renderer) => (
          <RendererProvider renderer={renderer} targetDocument={externalDocument}>
            <FluentProvider className={styles.provider} targetDocument={externalDocument}>
              <Example>
                Content rendered <b>within</b> iframe
              </Example>
            </FluentProvider>
          </RendererProvider>
        )}
      </FrameRenderer>
    </>
  );
};
```

Reply:
Thank you so much for the extra effort writing this up!